server/background_tasks.go

```diff
@@ -273,31 +300,32 @@ func (s *Server) uploadMsgsToTier2Storage() {
 				MaxAckPending: -1,
 				MaxDeliver:    10,
 			}
-			err := serv.memphisAddConsumer(tieredStorageStream, &cc)
+			err := serv.memphisAddConsumer(globalAccountName, tieredStorageStream, &cc)
 			if err != nil {
 				serv.Errorf("Failed add tiered storage consumer: " + err.Error())
 				return
 			}
 			TIERED_STORAGE_CONSUMER_CREATED = true
 		}
 		tieredStorageMapLock.Lock()
-		if len(tieredStorageMsgsMap.m) > 0 {
-			err := flushMapToTire2Storage()
-			if err != nil {
-				serv.Errorf("Failed upload messages to tiered 2 storage: " + err.Error())
-				tieredStorageMapLock.Unlock()
-				continue
-			}
+		err := flushMapToTire2Storage()
```

Comment:
Because in this function there is  a for loop - so if the map is empty - it is not enter to logical and thus the if is not relevant and I deleted the if @idanasulinmemphis